### DISCUSSION TITLE
migrate the WPF project From framework4.8 to core3.1 platform

### DISCUSSION THREAD

**Discussion by ZhengBT** [reactiveui/ReactiveUI#3541] [Q&A] [answered]
When I tried to migrate a WPF project, which referred the **ReactiveUI.WPF** NuGet package and was originally created in .NET Framework 4.8, to .NET Core 3.1 platform, an error occurred in the constructor of MainWindow.xaml.cs 
("[1]**The type initializer for 'ReactiveUI.ReactiveNotifyPropertyChangedMixin' threw an exception**"). After removing the ReactiveUI.WPF NuGet package, the error no longer existed in the constructor, but a new error occurred in this.WhenActivated ("[2]**Don't know how to detect when is activated/deactivated,you may need to implement IActivationForViewFetcher** "). Does this mean that ReactiveUI.WPF cannot run on the .NET Core 3.1 platform? Also, if I want to quickly migrate the project from .NET Framework 4.8 to a new platform, which platform version would be a better choice? I hope someone can help me clarify my confusion. Thank you!

**Comment by glennawatson** [comment, marked as answer]
.net core 3.1 is no longer supported by Microsoft. Upgrade to .net6 or above.

**Reply by ZhengBT** [reply]
Got it, I will try it out.Thank you! ^ - ^

**Comment by ChrisPulman** [comment]
NOTE: for the latest ReativeUI.Wpf with .Net 6 you will need to use the following minimum target net6.0-windows10.0.17763
